### PR TITLE
feat: edit goal modal inside goals tab

### PR DESCRIPTION
**Pull request by helloiambguedes**

### CODE REVIEW

File: app/packs/src/components-v2/pages/profile/profile-areas/goals/add-goal-modal/hooks/use-add-goal-modal.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { useCallback } from "react";
+import { useState } from "react";
+
+export const useAddGoalModalState = () => {
+  const [modalState, setModalState] = useState({ isOpen: false, type: "", item: undefined });
+  const openModal = useCallback((type, item) => setModalState({ isOpen: true, type, item }), [setModalState]);
```
Comment (francisco-leal) [inline app/packs/src/components-v2/pages/profile/profile-areas/goals/add-goal-modal/hooks/use-add-goal-modal.js:6]: is this state meant to start open?

File: app/packs/src/api/goals.js [hunk 2]
```diff
@@ -1,7 +1,20 @@
 import axios from "axios";
+import { appendCSRFToken, defaultHeaders } from "./utils";
 
 const getGoals = user => axios.get(`/api/v1/goals?id=${user}`);
 
+const deleteGoal = (user, goalId) => {
+  const baseHeaders = defaultHeaders();
+  const headers = appendCSRFToken(baseHeaders);
```
[the remark this answers is not in the document]
Comment (helloiambguedes) [reply app/packs/src/api/goals.js:8]: `current_user` is undefined on the BE without the CSRF token didn't really understand why

File: app/packs/src/components-v2/pages/profile/profile-areas/goals/add-goal-modal/hooks/use-add-goal-modal.js (hunk 1 above)
Comment (helloiambguedes) [reply app/packs/src/components-v2/pages/profile/profile-areas/goals/add-goal-modal/hooks/use-add-goal-modal.js:6]: Where?
It's false by default

`const [modalState, setModalState] = useState({ isOpen: false,...`

Comment (francisco-leal) [reply app/packs/src/components-v2/pages/profile/profile-areas/goals/add-goal-modal/hooks/use-add-goal-modal.js:6]: `const openModal = useCallback((type, item) => setModalState({ isOpen: true, type, item }), [setModalState]);`

**Pull request merged**